src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IEventAssignmentExpression.cs

```diff
@@ -967,21 +967,24 @@ void M(C c, EventHandler handler)
     Predecessors: [B0]
     Statements (1)
         IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null, IsInvalid) (Syntax: '(c.MyEvent  ... ndler) = 0;')
-          Expression: 
+          Expression:
             ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Void, IsInvalid) (Syntax: '(c.MyEvent  ... andler) = 0')
-              Left: 
+              Left:
                 IInvalidOperation (OperationKind.Invalid, Type: System.Void, IsInvalid, IsImplicit) (Syntax: 'c.MyEvent += handler')
                   Children(1):
                       IEventAssignmentOperation (EventAdd) (OperationKind.EventAssignment, Type: System.Void, IsInvalid) (Syntax: 'c.MyEvent += handler')
-                        Event Reference: 
+                        Event Reference:
                           IEventReferenceOperation: event System.EventHandler C.MyEvent (OperationKind.EventReference, Type: System.EventHandler, IsInvalid) (Syntax: 'c.MyEvent')
-                            Instance Receiver: 
+                            Instance Receiver:
                               IParameterReferenceOperation: c (OperationKind.ParameterReference, Type: C, IsInvalid) (Syntax: 'c')
-                        Handler: 
+                        Handler:
                           IParameterReferenceOperation: handler (OperationKind.ParameterReference, Type: System.EventHandler, IsInvalid) (Syntax: 'handler')
-              Right: 
-                ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 0) (Syntax: '0')
-
+              Right:
+                IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: System.Void, IsImplicit) (Syntax: '0')
```

Comment:
>System.Void [](http://example.com/codeflow?start=98&length=11)

Minor. Consider excluding `void` type as a meaningful target type for conversion.

Reply:
It looks like we call `GenerateConversionForAssignment` with `void` in other scenarios too (`VoidCall_ImplicitVoidConversion_DiscardAssignment` which is a conditional operator, and `CS0670ERR_FieldCantHaveVoidType01` which is a field initializer).
I'm leaning to leave as-is. But I'd also be okay to change and file an issue to align/tighten things up.